src/regex/match_data.cr

```diff
@@ -350,6 +346,16 @@ class Regex
       io << ')'
     end
 
+    # Prints the matched substring to *io*.
+    #
+    # ```
+    # "Crystal".match!(/yst/).to_s         # => "yst"
+    # "Crystal".match!(/(y)(s)(?=t)/).to_s # => "ys"
+    # ```
+    def to_s(io : IO) : Nil
```

Comment:
Am I supposed to also overwrite the no argument to_s overload? I got a little confused because some classes (like Nil) do, while others don't. It seems to work correctly either way

Reply:
Great observation! Usually you do not need to touch `to_s(IO)`. But here actually, yes.
The generic implementation of `#to_s` delegates to `#to_s(IO)` by wrapping it in `String.build`.
When `#to_s` is just a string that already exists, there's no need for a string builder and we can just return it directly, thus the need ro override `#to_s`.
This is the case here. A very similar example would be `Path#to_s` and we should do just the same here.